### PR TITLE
Backport of the Slider from the FiveM repo

### PR DESCRIPTION
**Pull request by justalemon**
This is a work in progress for the Slider item added on b5977ffe4a0c9de4daf42a5fd3f32e3833c1adcd at the FiveM repo.

### CODE REVIEW

**Comment by justalemon** [comment]
@Guad can you test this and let me know what changes it needs.

File: NativeUI/Items/UIMenuSliderItem.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            get
+            {
+                return _max;
+            }
+            set
+            {
+                if (_value > value)
+                {
+                    _value = value;
+                }
```
Comment (Guad) [inline NativeUI/Items/UIMenuSliderItem.cs]: Should set _max = value;

File: NativeUI/Items/UIMenuSliderItem.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        public int Value
+        {
+            get
+            {
+                return _value;
+            }
+            set
+            {
+                if (value > _max)
+                    _value = 100;
```
Comment (Guad) [inline NativeUI/Items/UIMenuSliderItem.cs]: Should be _value = _max;

File: NativeUI/Items/UIMenuSliderItem.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                else
```
Comment (Guad) [inline NativeUI/Items/UIMenuSliderItem.cs]: Also should probably check if _value < 0

File: NativeUI/UIMenu.cs [hunk 4]
```diff
@@ -1447,6 +1477,15 @@ protected virtual void ItemSelect(UIMenuItem selecteditem, int index)
             OnItemSelect?.Invoke(this, selecteditem, index);
         }
 
+        protected virtual void SliderChange(UIMenuSliderItem sender, int newindex)
+        {
+            OnSliderChange?.Invoke(this, sender, newindex);
+        }
+
+        protected virtual void SliderSelect(UIMenuSliderItem sender, int newindex)
```
Comment (Guad) [inline NativeUI/UIMenu.cs]: This is never called

File: NativeUI/Items/UIMenuSliderItem.cs [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        
+        protected int _value = 0;
+        protected int _max = 100;
+        protected int _multiplier = 5;
+
+
+        /// <summary>
+        /// Triggered when the slider is changed.
+        /// </summary>
+        public event ItemSliderEvent OnSliderChanged;
```
Comment (Guad) [inline NativeUI/Items/UIMenuSliderItem.cs]: Maybe add the select event here too?

File: NativeUI/Items/UIMenuSliderItem.cs (hunk 1 above)
Comment (justalemon) [reply NativeUI/Items/UIMenuSliderItem.cs]: I forgot about that. Fixed on 23b7dec150308f996249f81eda04fcdd45292f65.

File: NativeUI/Items/UIMenuSliderItem.cs (hunk 2 above)
Comment (justalemon) [reply NativeUI/Items/UIMenuSliderItem.cs]: This is a leftover when the max value was fixed to 100. Done in 41a50b620d5f11938f5c94edf7899f8824e40910.

File: NativeUI/Items/UIMenuSliderItem.cs (hunk 3 above)
Comment (justalemon) [reply NativeUI/Items/UIMenuSliderItem.cs]: Added an under zero check on 296b14754373514e9eed719665f3743d338c66e7.

File: NativeUI/UIMenu.cs (hunk 4 above)
Comment (justalemon) [reply NativeUI/UIMenu.cs]: There is UIMenuItem.Activated for that. It has been removed on 9f30dbbd272bce37699873d5d4019b99176603d6.

File: NativeUI/Items/UIMenuSliderItem.cs (hunk 5 above)
Comment (justalemon) [reply NativeUI/Items/UIMenuSliderItem.cs]: See above.

**Comment by justalemon** [comment]
Requested changes have been completed.

**Comment by Guad** [comment]
Looks good. I'll test it later tonight ingame

**Pull request merged**